src/azul/drs.py

```diff
@@ -93,26 +248,18 @@ def _uri_to_url(self, drs_uri: str, access_id: Optional[str] = None) -> str:
         """
         Translate a DRS URI into a DRS URL. All query params included in the DRS
         URI (eg '{drs_uri}?version=123') will be carried over to the DRS URL.
-        Only hostname-based DRS URIs (drs://<hostname>/<id>) are supported while
-        compact, identifier-based URIs (drs://[provider_code/]namespace:accession)
-        are not.
+        Compact, identifier-based URIs (drs://[provider_code/]namespace:accession)
+        are expanded to hostname-based DRS URIs (drs://<hostname>/<id>).
         """
-        parsed = furl(drs_uri)
-        scheme = 'drs'
-        require(parsed.scheme == scheme,
-                f'The URI {drs_uri!r} does not have the {scheme!r} scheme')
+        parsed = DRSURI.parse(drs_uri)
+        drs_url = parsed.to_url(self)
         # "The colon character is not allowed in a hostname-based DRS URI".
         # https://ga4gh.github.io/data-repository-service-schemas/preview/develop/docs/#_drs_uris
-        # It is worth noting that compact identifier-based URI can be hard to
-        # parse when following RFC3986, with the 'namespace:accession' part
-        # matching either the heir-part or path production depending if the
-        # optional provider code and following slash is included.
-        reject(':' in parsed.netloc or ':' in str(parsed.path),
-               f'The DRS URI {drs_uri!r} is not hostname-based')
-        parsed.scheme = 'https'
-        object_id = one(parsed.path.segments)
-        parsed.path.set(drs_object_url_path(object_id=object_id, access_id=access_id))
-        return str(parsed)
+        reject(':' in drs_url.netloc or ':' in str(drs_url.path),
+               f'The DRS URI {drs_url!r} is not hostname-based')
```

Comment:
This makes no sense. The error message doesn't match the code (URI vs url).

Also, try to encapsulate the code in this method in the classes of the DRSURI hierarchy so that the DRS URI handling code is consolidated there.